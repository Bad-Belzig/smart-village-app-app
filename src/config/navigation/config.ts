import { LinkingOptions } from '@react-navigation/native';
import * as Linking from 'expo-linking';

import { NavigatorConfig, ScreenName } from '../../types';
import { consts } from '../consts';

const { HOST_NAMES } = consts;

export const navigationConfig = (navigationType: 'drawer' | 'tab') => {
  let navigatorConfig: NavigatorConfig;
  const linkingConfig: LinkingOptions<{}> = {
    prefixes: [Linking.createURL('/')]
  };
  const screens = {
    [ScreenName.EncounterUserDetail]: HOST_NAMES.ENCOUNTER,
    [ScreenName.Detail]: HOST_NAMES.DETAIL,
    [ScreenName.Home]: '*'
  };

  if (navigationType != 'drawer') {
    // TODO: needs to be the index of the tab config where in-app links should navigate in
    const index = 0;

    navigatorConfig = {
      type: 'tab'
    };

    linkingConfig.config = {
      screens: {
        // For tab navigation choose the preferred tab by its position in the config array
        [`Stack${index}`]: {
          // The initialRouteName has to be the initial route of the chosen stack:
          // whatever is specified in the tab config for tab navigation
          initialRouteName: ScreenName.Home,
          screens
        }
      }
    };
  } else {
    navigatorConfig = {
      type: 'drawer'
    };

    linkingConfig.config = {
      screens: {
        AppStack: {
          // The initialRouteName has to be the initial route of the chosen stack:
          // Home for drawer navigation
          initialRouteName: ScreenName.Home,
          screens
        }
      }
    };
  }

  return { navigatorConfig, linkingConfig };
};
