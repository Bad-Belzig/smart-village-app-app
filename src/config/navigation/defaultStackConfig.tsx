import React from 'react';

import { HeaderLeft } from '../../components';
import { getScreenOptions } from '../../navigation/screenOptions';
import {
  AboutScreen,
  BookmarkCategoryScreen,
  BookmarkScreen,
  ConstructionSiteDetailScreen,
  ConstructionSiteOverviewScreen,
  DataProviderScreen,
  DetailScreen,
  EncounterDataScreen,
  EncounterHomeScreen,
  EncounterRegistrationScreen,
  EncounterScannerScreen,
  EncounterUserDetailScreen,
  FeedbackScreen,
  getTilesScreen,
  HomeScreen,
  HtmlScreen,
  IndexScreen,
  LunchScreen,
  MultiButtonScreen,
  NestedInfoScreen,
  OParlCalendarScreen,
  OParlDetailScreen,
  OParlOrganizationsScreen,
  OParlOverviewScreen,
  OParlPersonsScreen,
  OParlSearchScreen,
  SettingsScreen,
  SurveyDetailScreen,
  SurveyOverviewScreen,
  VolunteerDetailScreen,
  VolunteerIndexScreen,
  VolunteerScreen,
  WasteCollectionScreen,
  WasteReminderScreen,
  WeatherScreen,
  WebScreen
} from '../../screens';
import {
  DetailScreen as BBBUSDetailScreen,
  IndexScreen as BBBUSIndexScreen
} from '../../screens/BB-BUS';
import { ScreenName, StackConfig } from '../../types';
import { consts } from '../consts';
import { texts } from '../texts';

const { MATOMO_TRACKING } = consts;

export const defaultStackConfig = ({
  initialRouteName,
  isDrawer
}: {
  initialRouteName: ScreenName;
  isDrawer: boolean;
}): StackConfig => ({
  initialRouteName,
  screenOptions: getScreenOptions({ withDrawer: isDrawer }),
  screenConfigs: [
    {
      routeName: ScreenName.About,
      screenComponent: AboutScreen,
      screenOptions: { title: texts.screenTitles.about }
    },
    {
      routeName: ScreenName.BBBUSIndex,
      screenComponent: BBBUSIndexScreen
    },
    {
      routeName: ScreenName.BBBUSDetail,
      screenComponent: BBBUSDetailScreen,
      screenOptions: getScreenOptions({ withDrawer: isDrawer, withShare: true })
    },
    {
      routeName: ScreenName.BookmarkCategory,
      screenComponent: BookmarkCategoryScreen
    },
    {
      routeName: ScreenName.Bookmarks,
      screenComponent: BookmarkScreen,
      screenOptions: getScreenOptions({ withDrawer: isDrawer })
    },
    {
      routeName: ScreenName.Category,
      screenComponent: IndexScreen
    },
    {
      routeName: ScreenName.Company,
      screenComponent: getTilesScreen({
        matomoString: MATOMO_TRACKING.SCREEN_VIEW.COMPANY,
        staticJsonName: 'homeCompanies',
        titleFallback: texts.homeTitles.company,
        titleKey: 'headlineCompany'
      }),
      screenOptions: { title: texts.screenTitles.company }
    },
    {
      routeName: ScreenName.ConstructionSiteDetail,
      screenComponent: ConstructionSiteDetailScreen,
      screenOptions: { title: texts.screenTitles.constructionSite }
    },
    {
      routeName: ScreenName.ConstructionSiteOverview,
      screenComponent: ConstructionSiteOverviewScreen
    },
    {
      routeName: ScreenName.DataProvider,
      screenComponent: DataProviderScreen
    },
    {
      routeName: ScreenName.Detail,
      screenComponent: DetailScreen,
      screenOptions: getScreenOptions({ withDrawer: isDrawer, withBookmark: true, withShare: true })
    },
    {
      routeName: ScreenName.EncounterData,
      screenComponent: EncounterDataScreen,
      screenOptions: { title: texts.screenTitles.encounterHome }
    },
    {
      routeName: ScreenName.EncounterHome,
      screenComponent: EncounterHomeScreen,
      screenOptions: { title: texts.screenTitles.encounterHome }
    },
    {
      routeName: ScreenName.EncounterRegistration,
      screenComponent: EncounterRegistrationScreen,
      screenOptions: { title: texts.screenTitles.encounterHome }
    },
    {
      routeName: ScreenName.EncounterScanner,
      screenComponent: EncounterScannerScreen,
      screenOptions: { title: texts.screenTitles.encounterHome }
    },
    {
      routeName: ScreenName.EncounterUserDetail,
      screenComponent: EncounterUserDetailScreen,
      screenOptions: ({ navigation, route }) => ({
        title: texts.screenTitles.encounterHome,
        headerLeft: () => (
          <HeaderLeft
            onPress={() => {
              // @ts-expect-error we are lacking proper param types here
              if (route.params?.qrId || route.params?.fromPoll) {
                navigation.goBack();
              } else {
                navigation.navigate(ScreenName.EncounterHome);
              }
            }}
          />
        )
      })
    },
    {
      routeName: ScreenName.Form,
      screenComponent: FeedbackScreen,
      screenOptions: { title: texts.screenTitles.feedback }
    },
    {
      routeName: ScreenName.Feedback,
      screenComponent: FeedbackScreen,
      screenOptions: { title: texts.screenTitles.feedback }
    },
    {
      routeName: ScreenName.Home,
      screenComponent: HomeScreen,
      screenOptions: getScreenOptions({ withDrawer: isDrawer, withFavorites: true }),
      inititalParams: {
        isDrawer,
        title: texts.screenTitles.home
      }
    },
    {
      routeName: ScreenName.Html,
      screenComponent: HtmlScreen
    },
    {
      routeName: ScreenName.Index,
      screenComponent: IndexScreen
    },
    {
      routeName: ScreenName.Lunch,
      screenComponent: LunchScreen
    },
    {
      routeName: ScreenName.MultiButton,
      screenComponent: MultiButtonScreen
    },
    {
      routeName: ScreenName.NestedInfo,
      screenComponent: NestedInfoScreen
    },
    {
      routeName: ScreenName.OParlCalendar,
      screenComponent: OParlCalendarScreen
    },
    {
      routeName: ScreenName.OParlDetail,
      screenComponent: OParlDetailScreen
    },
    {
      routeName: ScreenName.OParlOrganizations,
      screenComponent: OParlOrganizationsScreen
    },
    {
      routeName: ScreenName.OParlOverview,
      screenComponent: OParlOverviewScreen
    },
    {
      routeName: ScreenName.OParlPersons,
      screenComponent: OParlPersonsScreen
    },
    {
      routeName: ScreenName.OParlSearch,
      screenComponent: OParlSearchScreen
    },
    {
      routeName: ScreenName.Service,
      screenComponent: getTilesScreen({
        matomoString: MATOMO_TRACKING.SCREEN_VIEW.SERVICE,
        staticJsonName: 'homeService',
        titleFallback: texts.homeTitles.service,
        titleKey: 'headlineService'
      }),
      screenOptions: { title: texts.screenTitles.service }
    },
    {
      routeName: ScreenName.Settings,
      screenComponent: SettingsScreen,
      screenOptions: { title: texts.screenTitles.settings }
    },
    {
      routeName: ScreenName.SurveyDetail,
      screenComponent: SurveyDetailScreen,
      screenOptions: { title: texts.screenTitles.survey }
    },
    {
      routeName: ScreenName.SurveyOverview,
      screenComponent: SurveyOverviewScreen,
      screenOptions: { title: texts.screenTitles.surveys }
    },
    {
      routeName: ScreenName.VolunteerDetail,
      screenComponent: VolunteerDetailScreen
    },
    {
      routeName: ScreenName.VolunteerIndex,
      screenComponent: VolunteerIndexScreen
    },
    {
      routeName: ScreenName.Volunteer,
      screenComponent: VolunteerScreen,
      screenOptions: getScreenOptions({ withVolunteer: true }),
      inititalParams: {
        title: texts.screenTitles.volunteer.home
      }
    },
    {
      routeName: ScreenName.WasteCollection,
      screenComponent: WasteCollectionScreen,
      screenOptions: { title: texts.screenTitles.wasteCollection }
    },
    {
      routeName: ScreenName.WasteReminder,
      screenComponent: WasteReminderScreen,
      screenOptions: { title: texts.screenTitles.wasteCollection }
    },
    {
      routeName: ScreenName.Weather,
      screenComponent: WeatherScreen
    },
    {
      routeName: ScreenName.Web,
      screenComponent: WebScreen,
      screenOptions: getScreenOptions({ withDrawer: isDrawer, withShare: true })
    }
  ]
});
