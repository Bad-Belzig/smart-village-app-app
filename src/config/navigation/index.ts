export * from './config';
export * from './defaultStackConfig';
