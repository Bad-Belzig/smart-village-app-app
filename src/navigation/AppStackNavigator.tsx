import { createStackNavigator } from '@react-navigation/stack';
import React from 'react';

import { StackConfig } from '../types';

const Stack = createStackNavigator<Record<string, { title: string } | undefined>>();

export const getStackNavigator = (stackConfig: StackConfig) => () =>
  (
    <Stack.Navigator
      initialRouteName={stackConfig.initialRouteName}
      screenOptions={stackConfig.screenOptions}
    >
      {stackConfig.screenConfigs.map((screenConfig) => (
        <Stack.Screen
          key={screenConfig.routeName}
          name={screenConfig.routeName}
          component={screenConfig.screenComponent}
          options={screenConfig.screenOptions}
          initialParams={screenConfig.inititalParams}
        />
      ))}
    </Stack.Navigator>
  );
