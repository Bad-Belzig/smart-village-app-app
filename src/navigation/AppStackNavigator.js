import { createStackNavigator } from '@react-navigation/stack';
import React from 'react';

import { texts } from '../config';
import {
  BookmarkCategoryScreen,
  BookmarkScreen,
  ConstructionSiteDetailScreen,
  ConstructionSiteOverviewScreen,
  DataProviderScreen,
  DetailScreen,
  FormScreen,
  HomeScreen,
  HtmlScreen,
  IndexScreen,
  LunchScreen,
  OParlCalendarScreen,
  OParlDetailScreen,
  OParlOrganizationsScreen,
  OParlOverviewScreen,
  OParlPersonsScreen,
  OParlSearchScreen,
  SettingsScreen,
  SurveyDetailScreen,
  SurveyOverviewScreen,
  WasteCollectionScreen,
  WasteReminderScreen,
  WeatherScreen,
  WebScreen
} from '../screens';
import {
  DetailScreen as BBBUSDetailScreen,
  IndexScreen as BBBUSIndexScreen
} from '../screens/BB-BUS';
import {
  defaultStackNavigatorScreenOptions,
  detailScreenOptions,
  homeScreenOptions,
  screenOptionsWithSettings,
  screenOptionsWithShare
} from './screenOptions';

const Stack = createStackNavigator();

export const AppStackNavigator = (headerRight = true) => {
  return (
    <Stack.Navigator
      initialRouteName="Home"
      screenOptions={defaultStackNavigatorScreenOptions(headerRight)}
    >
      <Stack.Screen name="BBBUSIndex" component={BBBUSIndexScreen} />
      <Stack.Screen
        name="BBBUSDetail"
        component={BBBUSDetailScreen}
        options={screenOptionsWithShare(headerRight)}
      />
      <Stack.Screen
        name="Bookmarks"
        component={BookmarkScreen}
        options={screenOptionsWithSettings(headerRight)}
      />
      <Stack.Screen name="BookmarkCategory" component={BookmarkCategoryScreen} />
      <Stack.Screen name="Category" component={IndexScreen} />
      <Stack.Screen name="ConstructionSiteDetail" component={ConstructionSiteDetailScreen} />
      <Stack.Screen name="ConstructionSiteOverview" component={ConstructionSiteOverviewScreen} />
      <Stack.Screen name="DataProvider" component={DataProviderScreen} />
      <Stack.Screen
        name="Detail"
        component={DetailScreen}
        options={detailScreenOptions(headerRight)}
      />
      <Stack.Screen name="Form" component={FormScreen} />
      <Stack.Screen name="Home" component={HomeScreen} options={homeScreenOptions(headerRight)} />
      <Stack.Screen name="Html" component={HtmlScreen} />
      <Stack.Screen name="Index" component={IndexScreen} />
      <Stack.Screen name="Lunch" component={LunchScreen} />
      <Stack.Screen name="OParlCalendar" component={OParlCalendarScreen} />
      <Stack.Screen name="OParlDetail" component={OParlDetailScreen} />
      <Stack.Screen name="OParlOrganizations" component={OParlOrganizationsScreen} />
      <Stack.Screen name="OParlOverview" component={OParlOverviewScreen} />
      <Stack.Screen name="OParlPersons" component={OParlPersonsScreen} />
      <Stack.Screen name="OParlSearch" component={OParlSearchScreen} />
      <Stack.Screen
        name="Settings"
        component={SettingsScreen}
        options={{ title: texts.screenTitles.settings }}
      />
      <Stack.Screen
        name="SurveyDetail"
        component={SurveyDetailScreen}
        options={{ title: texts.screenTitles.survey }}
      />
      <Stack.Screen
        name="SurveyOverview"
        component={SurveyOverviewScreen}
        options={{ title: texts.screenTitles.surveys }}
      />
      <Stack.Screen
        name="WasteCollection"
        component={WasteCollectionScreen}
        options={{ title: texts.screenTitles.wasteCollection }}
      />
      <Stack.Screen
        name="WasteReminder"
        component={WasteReminderScreen}
        options={{ title: texts.screenTitles.wasteCollection }}
      />
      <Stack.Screen name="Weather" component={WeatherScreen} />
      <Stack.Screen name="Web" component={WebScreen} />
    </Stack.Navigator>
  );
};
