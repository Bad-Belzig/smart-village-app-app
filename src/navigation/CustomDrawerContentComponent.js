import PropTypes from 'prop-types';
import React, { useContext } from 'react';
import { Platform, StyleSheet, TouchableOpacity, View } from 'react-native';
import { Icon } from 'react-native-elements';
import { getStatusBarHeight } from 'react-native-status-bar-height';

import { colors, device } from '../config';
import { DiagonalGradient } from '../components';
import DrawerNavigatorItems from './DrawerNavigatorItems';
import { OrientationContext } from '../OrientationProvider';

/**
 * based on the default content component from React Navigation:
 *   https://github.com/react-navigation/drawer/blob/c5954d744f463e7f1c67941b8eb6914c0101e56c/src/navigators/createDrawerNavigator.tsx
 */
// FIXME: Nav
export const CustomDrawerContentComponent = ({ navigation, drawerRoutes, state }) => {
  const { orientation } = useContext(OrientationContext);

  return (
    <DiagonalGradient>
      <View style={styles.headerContainer}>
        <View style={stylesWithProps({ orientation }).header}>
          <TouchableOpacity
            accessibilityLabel="Schließen Taste"
            accessibilityHint="Menü schließen"
            onPress={() => navigation.closeDrawer()}
            delayPressIn={0}
          >
            <Icon
              name="ios-close"
              type="ionicon"
              color={colors.lightestText}
              containerStyle={stylesWithProps({ orientation }).icon}
            />
          </TouchableOpacity>
        </View>
      </View>
      <DrawerNavigatorItems navigation={navigation} state={state} drawerRoutes={drawerRoutes} />
    </DiagonalGradient>
  );
};

const styles = StyleSheet.create({
  headerContainer: Platform.select({
    android: {
      backgroundColor: colors.darkText,
      elevation: 2
    },
    ios: {
      shadowColor: colors.darkText,
      shadowOffset: { height: 1, width: 0 },
      shadowOpacity: 0.5,
      shadowRadius: 3
    }
  })
});

export const getHeaderHeight = (orientation) =>
  // Android always 56
  // iOS:
  //   portrait: 44
  //   landscape: tablet 66 / phone 32
  Platform.select({
    android: 56,
    ios: orientation === 'landscape' ? (!Platform.isPad ? 32 : 66) : 44
  });

export const statusBarHeight = (orientation) =>
  device.platform === 'android'
    ? getStatusBarHeight()
    : orientation === 'portrait'
    ? getStatusBarHeight()
    : 0;

/* eslint-disable react-native/no-unused-styles */
/* this works properly, we do not want that warning */
const stylesWithProps = ({ orientation }) => {
  const height = getHeaderHeight(orientation) + statusBarHeight(orientation);

  return StyleSheet.create({
    header: {
      alignItems: 'flex-end',
      flexDirection: 'row',
      backgroundColor: colors.primary,
      height,
      justifyContent: 'flex-end'
    },
    icon: {
      paddingHorizontal: 24,
      ...Platform.select({
        android: {
          paddingVertical: 17
        },
        ios: {
          paddingVertical: orientation === 'landscape' && !Platform.isPad ? 5 : 8
        }
      })
    }
  });
};
/* eslint-enable react-native/no-unused-styles */

CustomDrawerContentComponent.propTypes = {
  drawerRoutes: PropTypes.object.isRequired,
  navigation: PropTypes.object.isRequired,
  state: PropTypes.object.isRequired
};
