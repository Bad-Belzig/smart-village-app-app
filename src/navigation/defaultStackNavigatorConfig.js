import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import CardStackStyleInterpolator from 'react-navigation-stack/src/views/StackView/StackViewStyleInterpolator';

import { colors, consts, device, normalize } from '../config';
import { DiagonalGradient, Icon } from '../components';
import { drawerMenu } from '../icons';

export const defaultStackNavigatorConfig = (initialRouteName, headerRight = true) => {
  return {
    initialRouteName,
    URIPrefix: 'smart-village-app://',
    defaultNavigationOptions: ({ navigation }) => ({
      // header gradient:
      // https://stackoverflow.com/questions/44924323/react-navigation-gradient-color-for-header
      headerBackground: <DiagonalGradient />,
      headerTitleStyle: {
        color: colors.lightestText,
        fontFamily: device.platform === 'ios' ? 'titillium-web-bold' : 'titillium-web-regular',
        fontSize: normalize(20),
        fontWeight: '400',
        lineHeight: normalize(29)
      },
      headerTitleContainerStyle: {
        flexDirection: 'row',
        justifyContent: 'flex-start'
      },
      headerRight: headerRight && (
        <TouchableOpacity
          onPress={() => navigation.openDrawer()}
          accessibilityLabel={consts.a11yLabel.openMenuIcon}
          accessibilityHint={consts.a11yLabel.menuHint}
        >
          <Icon xml={drawerMenu(colors.lightestText)} style={styles.icon} />
        </TouchableOpacity>
      )
    }),
    transitionConfig: () => ({
      screenInterpolator: (sceneProps) => {
        return CardStackStyleInterpolator.forHorizontal(sceneProps);
      }
    })
  };
};

const styles = StyleSheet.create({
  icon: {
    paddingHorizontal: normalize(14),
    paddingVertical: normalize(4)
  }
});
