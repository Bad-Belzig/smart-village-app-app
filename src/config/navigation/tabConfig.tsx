/* eslint-disable react-native/no-inline-styles */
import React from 'react';

import { OrientationAwareIcon } from '../../components';
import { ScreenName, TabConfig, TabNavigatorConfig } from '../../types';
import { colors } from '../colors';
import { Icon } from '../icons';
import { normalize } from '../normalize';
import { texts } from '../texts';

import { defaultStackConfig } from './defaultStackConfig';

type TabBarIconProps = {
  focused: boolean;
  color: string;
  size: number;
};

const homeTabConfig: TabConfig = {
  stackConfig: defaultStackConfig({
    initialRouteName: ScreenName.Home,
    isDrawer: false
  }),
  tabOptions: {
    tabBarAccessibilityLabel: `${texts.tabBarLabel.home} (Tab 1 von 5)`,
    tabBarLabel: texts.tabBarLabel.home,
    tabBarIcon: ({ color }: TabBarIconProps) => (
      <OrientationAwareIcon color={color} Icon={Icon.Home} />
    )
  }
};

const serviceTabConfig: TabConfig = {
  stackConfig: defaultStackConfig({
    initialRouteName: ScreenName.Service,
    isDrawer: false
  }),
  tabOptions: {
    tabBarLabel: texts.tabBarLabel.service,
    tabBarIcon: ({ color }: TabBarIconProps) => (
      <OrientationAwareIcon color={color} Icon={Icon.Service} />
    )
  }
};

const pointOfInterestTabConfig: TabConfig = {
  stackConfig: defaultStackConfig({
    initialRouteName: ScreenName.Index,
    isDrawer: false
  }),
  tabOptions: {
    tabBarLabel: texts.tabBarLabel.pointsOfInterest,
    tabBarIcon: ({ color }: TabBarIconProps) => (
      <OrientationAwareIcon color={color} Icon={Icon.Location} size={normalize(30)} />
    )
  }
};

const eventsTabConfig: TabConfig = {
  stackConfig: defaultStackConfig({
    initialRouteName: ScreenName.Events,
    isDrawer: false
  }),
  tabOptions: {
    tabBarLabel: texts.tabBarLabel.events,
    tabBarIcon: ({ color }: TabBarIconProps) => (
      <OrientationAwareIcon color={color} Icon={Icon.Calendar} size={normalize(24)} />
    )
  }
};

const aboutTabConfig: TabConfig = {
  stackConfig: defaultStackConfig({
    initialRouteName: ScreenName.About,
    isDrawer: false
  }),
  tabOptions: {
    tabBarAccessibilityLabel: `${texts.tabBarLabel.about} (Tab 5 von 5)`,
    tabBarLabel: texts.tabBarLabel.about,
    tabBarIcon: ({ color }: TabBarIconProps) => (
      <OrientationAwareIcon
        color={color}
        Icon={Icon.About}
        landscapeStyle={{ marginRight: -normalize(6) }}
        size={normalize(28)}
        style={{ marginTop: normalize(3) }}
      />
    )
  }
};

export const tabNavigatorConfig: TabNavigatorConfig = {
  activeTintColor: colors.accent,
  inactiveTintColor: colors.primary,
  activeBackgroundColor: colors.surface,
  inactiveBackgroundColor: colors.surface,
  tabConfigs: [
    homeTabConfig,
    serviceTabConfig,
    pointOfInterestTabConfig,
    eventsTabConfig,
    aboutTabConfig
  ]
};
