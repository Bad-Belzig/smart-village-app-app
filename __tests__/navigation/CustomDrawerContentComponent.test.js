import { CustomDrawerContentComponent } from '../../src/navigation/CustomDrawerContentComponent';

describe('CustomDrawerContentComponent', () => {
  it('creates correctly', () => {
    expect(CustomDrawerContentComponent).toMatchSnapshot();
  });
});
