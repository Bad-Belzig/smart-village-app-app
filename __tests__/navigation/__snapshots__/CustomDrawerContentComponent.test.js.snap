// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomDrawerContentComponent creates correctly 1`] = `[Function]`;
